scripts/cypress_run.py

```diff
@@ -0,0 +1,146 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import hashlib
+import os
+import subprocess
+from datetime import datetime
+
+XVFB_PRE_CMD = "xvfb-run --auto-servernum --server-args='-screen 0, 1024x768x24' "
+REPO = os.getenv("GITHUB_REPOSITORY") or "apache/superset"
+GITHUB_EVENT_NAME = os.getenv("GITHUB_REPOSITORY") or "push"
+
+
+def compute_hash(file_path: str) -> str:
+    return hashlib.md5(file_path.encode()).hexdigest()
+
+
+def compute_group_index(hash_value: str, num_groups: int) -> int:
+    return int(hash_value, 16) % num_groups
+
+
+def generate_build_id() -> str:
+    now = datetime.now()
+    rounded_minute = now.minute - (now.minute % 20)
+    rounded_time = now.replace(minute=rounded_minute, second=0, microsecond=0)
+    return (os.getenv("GITHUB_SHA") or "DUMMY")[:8] + rounded_time.strftime(
+        "%Y%m%d%H%M"
+    )
+
+
+def get_cypress_cmd(
+    spec_list: list[str], _filter: str, group: str, use_dashboard: bool
+) -> str:
+    cypress_cmd = "./node_modules/.bin/cypress run"
+
+    os.environ["TERM"] = "xterm"
+    os.environ["ELECTRON_DISABLE_GPU"] = "true"
+    build_id = generate_build_id()
+    browser = os.getenv("CYPRESS_BROWSER", "chrome")
+
+    if use_dashboard:
+        # Run using cypress.io service
+        cypress_record_key = (
+            subprocess.check_output(
+                ["echo", os.getenv("CYPRESS_KEY") or "DUMMY", "|", "base64", "--decode"]
```

Comment:
Whenever there's an action that runs a script, and the script uses a KEY, it raises my security flag now :) CC @dpgaspar to make sure this is OK :)

Reply:
this is triggered by `pull_request` GHA so should be ok, and I'm assuming `CYPRESS_KEY` is not a secret right?
yes script can be "abused" by a PR but will it have privileges to do anything destructive?

Reply:
I think not, I didn't realize the key is public.